db_stress_tool/expected_state.cc

```diff
@@ -443,13 +473,56 @@ class ExpectedStateTraceRecordHandler : public TraceRecord::Handler,
                  const Slice& value) override {
     Slice key =
         StripTimestampFromUserKey(key_with_ts, FLAGS_user_timestamp_size);
+
+    bool should_buffer_write = (buffered_writes_ != nullptr);
+    if (should_buffer_write) {
+      return WriteBatchInternal::Merge(buffered_writes_, column_family_id, key,
+                                       value);
+    }
+
     return PutCF(column_family_id, key, value);
```

Comment:
[LearningQuestion]
Is there any reason why PutCF is used instead of MergeCF(column_family_id, key_with_ts, value)???

Reply:
These are the possible merge operators used by `db_stress`, which make the `Merge()` look the same as a `Put()` from the verification's perspective: https://github.com/facebook/rocksdb/blob/5fbcc8c54d4a8704405b568d53faf23cd0722eeb/db_stress_tool/db_stress_test_base.cc#L3120-L3122